src/prof.c

```diff
@@ -1756,6 +1756,9 @@ prof_fdump(void) {
 		return;
 	}
 	tsd = tsd_fetch();
+	if (tsd->reentrancy_level > 0) {
```

Comment:
I don't think it's possible to trigger this, because we only call this function via an ```atexit(3)``` callback.  An assertion should be sufficient.